x-pack/solutions/security/plugins/security_solution/public/asset_inventory/components/filters.tsx

```diff
@@ -0,0 +1,229 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState, useEffect, useMemo, useCallback } from 'react';
+import { EuiSpacer, EuiFlexItem } from '@elastic/eui';
+// import { AlertFilterControls } from '@kbn/alerts-ui-shared/src/alert_filter_controls';
+import { FilterGroup } from '@kbn/alerts-ui-shared/src/alert_filter_controls/filter_group';
+import type { FilterControlConfig } from '@kbn/alerts-ui-shared';
+import type { FieldSpec } from '@kbn/data-views-plugin/common';
+// import type { Filter } from '@kbn/es-query';
+import type { DataViewBase } from '@kbn/es-query';
+import { createKbnUrlStateStorage, Storage } from '@kbn/kibana-utils-plugin/public';
+import { ControlGroupRenderer } from '@kbn/controls-plugin/public';
+import type { DataViewSpec } from '@kbn/data-plugin/common';
+import { SECURITY_SOLUTION_RULE_TYPE_IDS } from '@kbn/securitysolution-rules';
+import { useHistory } from 'react-router-dom';
+import { useKibana } from '../../common/lib/kibana';
+import { useSpaceId } from '../../common/hooks/use_space_id';
+import { URL_PARAM_KEY } from '../../common/hooks/use_url_state';
+import { useGlobalTime } from '../../common/containers/use_global_time';
+import { FilterGroupLoading } from './filters_loading';
+
+const SECURITY_ASSET_INVENTORY_DATA_VIEW = {
+  id: 'security_solution_asset_inventory_dv',
+  name: 'Security Solution Asset Inventory DataView',
+};
+
+const DEFAULT_ASSET_INVENTORY_FILTERS: FilterControlConfig[] = [
+  {
+    title: 'Type',
+    fieldName: 'asset.category',
+  },
+  {
+    title: 'Criticality',
+    fieldName: 'asset.criticality',
+  },
+  {
+    title: 'Tags',
+    fieldName: 'asset.tags.name',
+  },
+  {
+    title: 'Name',
+    fieldName: 'asset.name', // TODO this is TBD
+  },
+];
+
+type DataView = Omit<DataViewBase, 'fields'> & { fields: FieldSpec[] };
+
+export interface FiltersProps {
+  // TODO Remove dataView prop when integrating with backend (fetched from hook)
+  dataView: DataView;
+  dataViewSpec: DataViewSpec;
+}
+
+export const Filters = ({ dataView, dataViewSpec: indexPattern, ...props }: FiltersProps) => {
+  const {
+    // http,
+    // notifications: { toasts },
+    dataViews,
+  } = useKibana().services;
+  const { to, from } = useGlobalTime();
+  const spaceId = useSpaceId();
+  const history = useHistory();
+  const urlStorage = useMemo(
+    () =>
+      createKbnUrlStateStorage({
+        history,
+        useHash: false,
+        useHashQuery: false,
+      }),
+    [history]
+  );
+  const filterControlsUrlState = useMemo(
+    () => urlStorage.get<FilterControlConfig[] | undefined>(URL_PARAM_KEY.pageFilter) ?? undefined,
+    [urlStorage]
+  );
+
+  const setFilterControlsUrlState = useCallback(
+    (newFilterControls: FilterControlConfig[]) => {
+      urlStorage.set(URL_PARAM_KEY.pageFilter, newFilterControls);
```

Comment:
I think we are going to have a conflict with the Alerts page if we reuse the same key `URL_PARAM_KEY.pageFilter`, it seems we are going to create our own key for `assetInventoryPage` on https://github.com/elastic/kibana/blob/da25d13a2ac5be97216e202bf5d6bda6ff920b3b/x-pack/solutions/security/plugins/security_solution/public/common/hooks/use_url_state.ts